### ISSUE TITLE
Dangerous default save_dir in convert_coco() in ultralytics.data.converter

### ISSUE THREAD

**Issue by Godush** [ultralytics/ultralytics#5350]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

The default save_dir parameter in the method convert_coco() is set to '.' which I find is dangerous, because this directory is deleted at the beginning of the function, so it ends up in deleting the script directory and the script itself if the parameter is not set. So it could lead to deleting a whole project directory if someone does not know this. So my suggestion would be using a different default save_dir or making it at least a required parameter.
Also the example in the documentation leads to this "dangerous" behaviour: https://docs.ultralytics.com/datasets/segment/ (COCO Dataset Format to YOLO Format)
Thank you and best regards

### Environment

```
Ultralytics YOLOv8.0.196  Python-3.11.2 torch-2.0.1+cpu CPU (11th Gen Intel Core(TM) i7-1165G7 2.80GHz)
Setup complete  (8 CPUs, 15.7 GB RAM, 273.8/476.2 GB disk)

OS                  Windows-10-10.0.22621-SP0
Environment         Windows
Python              3.11.2
Install             pip
RAM                 15.71 GB
CPU                 11th Gen Intel Core(TM) i7-1165G7 2.80GHz
CUDA                None
```

### Minimal Reproducible Example

```
from ultralytics.data.converter import convert_coco

# not passing a save_dir will result in deleting the current directory
convert_coco(labels_dir='labels', use_segments=True)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Godush** [comment]
Thanks a lot

**Comment by glenn-jocher** [comment]
@Godush thank you for bringing this issue to our attention. We certainly want to ensure the safety and integrity of users' data. Your suggestion is valid, and we plan on addressing it. Instead of having '.' as the default `save_dir` which indeed leads to the entire directory being deleted, we'll make it a required parameter. This way, users will have to consciously specify the directory. We'll also update our documentation to clearly indicate this requirement. Thanks again for your valuable feedback, it really helps improve the project!

**Comment by glenn-jocher** [comment]
@Godush great news 😃! Your original issue may now be resolved ✅ in PR #5350 and should be released in `ultralytics 8.0.197` soon.

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
